### PR TITLE
staticcheck tests pass

### PR DESCRIPTION
**Pull request by tharvik**
I've been playing with [staticcheck](https://staticcheck.io), a static code analyser for go (the best one/most complete out there IMO), and ran it on the codebase. This is a bunch of changes so I'm making a first pass for the `*_test.go` files which shouldn't change any behavior.

While I'm at it, I've also `sed` some inconsistent pattern to have nicer messages when encountering errors.

I still have some staticcheck errors, related to deprecated uses in tests. I didn't update theses as I wan't sure if it was wanted to check for theses code paths, but from the look of it, it's not explicit that its checking retro compatibility. Some could be change quite safely, such as using `Roster.GetID` instead of `Roster.ID` or moves to kyber, but some other, such as using bndproto instead of blscosi/protocol, might not be wanted. Here goes the list
```
blscosi/bdnproto/protocol_test.go:10:2: Package go.dedis.ch/cothority/v3/blscosi/protocol is deprecated: use the bdnproto instead to be robust against the rogue public-key attack described here: https://crypto.stanford.edu/~dabo/pubs/papers/BLSmultisig.html  (SA1019)
byzcoin/proof_test.go:137:6: from.Roster.ID is deprecated: ID can be generated using the list of server identities. Use GetID instead.  (SA1019)
byzcoin/proof_test.go:137:27: to.Roster.ID is deprecated: ID can be generated using the list of server identities. Use GetID instead.  (SA1019)
byzcoin/service_test.go:1696:19: latest.Roster.ID is deprecated: ID can be generated using the list of server identities. Use GetID instead.  (SA1019)
byzcoin/service_test.go:1696:42: rosterR.ID is deprecated: ID can be generated using the list of server identities. Use GetID instead.  (SA1019)
byzcoin/service_test.go:1715:19: latest.Roster.ID is deprecated: ID can be generated using the list of server identities. Use GetID instead.  (SA1019)
byzcoin/service_test.go:1715:42: rosterR.ID is deprecated: ID can be generated using the list of server identities. Use GetID instead.  (SA1019)
byzcoin/service_test.go:1769:19: latest.Roster.ID is deprecated: ID can be generated using the list of server identities. Use GetID instead.  (SA1019)
byzcoin/service_test.go:1769:42: rosterR.ID is deprecated: ID can be generated using the list of server identities. Use GetID instead.  (SA1019)
byzcoinx/byzcoinx_test.go:15:2: Package go.dedis.ch/cothority/v3/blscosi/protocol is deprecated: use the bdnproto instead to be robust against the rogue public-key attack described here: https://crypto.stanford.edu/~dabo/pubs/papers/BLSmultisig.html  (SA1019)
ftcosi/protocol/protocol_test.go:95:68: cosi.CompletePolicy is deprecated: the policy has moved to the package kyber/sign  (SA1019)
ftcosi/protocol/protocol_test.go:142:71: cosi.NewThresholdPolicy is deprecated: the policy has moved to the package kyber/sign  (SA1019)
ftcosi/protocol/protocol_test.go:222:68: cosi.NewThresholdPolicy is deprecated: the policy has moved to the package kyber/sign  (SA1019)
ftcosi/protocol/protocol_test.go:291:68: cosi.NewThresholdPolicy is deprecated: the policy has moved to the package kyber/sign  (SA1019)
ftcosi/protocol/protocol_test.go:470:57: cosi.CompletePolicy is deprecated: the policy has moved to the package kyber/sign  (SA1019)
ftcosi/protocol/protocol_test.go:476:57: cosi.NewThresholdPolicy is deprecated: the policy has moved to the package kyber/sign  (SA1019)
ftcosi/protocol/protocol_test.go:494:26: cosi.Policy is deprecated: the policies have moved to the package kyber/sign  (SA1019)
ftcosi/protocol/protocol_test.go:508:26: cosi.Policy is deprecated: the policies have moved to the package kyber/sign  (SA1019)
ftcosi/service/ftcosi_test.go:40:82: cosi.CompletePolicy is deprecated: the policy has moved to the package kyber/sign  (SA1019)
ftcosi/service/ftcosi_test.go:65:79: cosi.CompletePolicy is deprecated: the policy has moved to the package kyber/sign  (SA1019)
```

At some point, it would be great to add it to the CI, this is a great tool, but I'm not knowledgeable enough of the internals to do it safely. If someone else want to play with it, please do so :)

### CODE REVIEW

File: eventlog/api_test.go [hunk 1]
```diff
@@ -439,8 +439,8 @@ func TestClient_StreamEventsFrom(t *testing.T) {
 	}
 
 	wg := sync.WaitGroup{}
+	wg.Add(1)
 	go func() {
-		wg.Add(1)
```
Comment (nkcr) [inline eventlog/api_test.go]: why?

File: byzcoin/service_test.go [hunk 2]
```diff
@@ -519,8 +521,8 @@ func TestService_AutomaticVersionUpgrade(t *testing.T) {
 
 	closing := make(chan bool)
 	wg := sync.WaitGroup{}
+	wg.Add(1)
 	go func(closeChan chan bool) {
-		wg.Add(1)
```
[the remark this answers is not in the document]
Comment (tharvik) [reply byzcoin/service_test.go]: Yep, was weird for me at first.

The [staticcheck reference](https://staticcheck.io/docs/checks#SA2000) describe it as 
> sync.WaitGroup.Add called inside the goroutine, leading to a race condition

Which is correct by the [golang documentation](https://golang.org/pkg/sync/#WaitGroup.Add)
> Typically this means the calls to Add should execute before the statement creating the goroutine or other event to be waited for.

In this case, it's not really relevant, as there is a single go routine to wait for, the `sync.WaitGroup` is overkill, as using a chan would easily do the trick :P

File: eventlog/api_test.go (hunk 1 above)
Comment (tharvik) [reply eventlog/api_test.go]: Same idea

File: byzcoin/service_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (nkcr) [reply byzcoin/service_test.go]: nice, that's good to know!

**Comment by nkcr** [comment]
I'll merge it once the conflict is resolved

**Comment by tharvik** [comment]
It's resolved :)

**Pull request merged**